tests/phpunit/tests/blocks/getHookedBlockMarkup

```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * Tests for the get_hooked_block_markup function.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ *
+ * @since 6.5.0
+ *
+ * @group blocks
+ * @group block-hooks
+ */
+class Tests_Blocks_GetHookedBlockMarkup extends WP_UnitTestCase
+{
+	/**
+	 * @ticket 59646
+	 *
+	 * @covers ::get_hooked_block_markup
+	 */
+	public function test_get_hooked_block_markup_adds_metadata() {
+		$anchor_block = array(
+			'blockName' => 'tests/anchor-block',
+		);
+
+		$actual = get_hooked_block_markup( $anchor_block, 'tests/hooked-block' );
+		$this->assertSame( array( 'tests/hooked-block' ), $anchor_block['attrs']['metadata']['ignoredHookedBlocks'] );
+		$this->assertSame( '<!-- wp:tests/hooked-block /-->', $actual );
+	}
+
+	/**
+	 * @ticket 59646
+	 *
+	 * @covers ::get_hooked_block_markup
+	 */
+	public function test_get_hooked_block_markup_if_block_is_already_hooked() {
+		$anchor_block = array(
+			'blockName' => 'tests/anchor-block',
+			'attrs'     => array(
+				'metadata' => array(
+					'ignoredHookedBlocks' => array( 'tests/hooked-block' ),
+				),
+			)
+		);
+
+		$actual = get_hooked_block_markup( $anchor_block, 'tests/hooked-block' );
+		$this->assertSame( array( 'tests/hooked-block' ), $anchor_block['attrs']['metadata']['ignoredHookedBlocks'] );
+		$this->assertSame( '', $actual );
+	}
```

Comment:
One additional test could cover the case where the anchor block has some ignored hooked blocks listed and someone activates a new plugin that wants to inject more hooked blocks. In effect,  a combination of two existing tests which validates pushing a new value to an array.

Reply:
Sounds good, I'll add that!

Reply:
Excellent 👍